### PR TITLE
Adding advertising and publishing on the SceneManager.ts

### PR DESCRIPTION
**Pull request by liamhan0905**
### Related PR: 
- https://github.com/osrf/pounce-web/pull/17
- https://github.com/gazebosim/gz-sim/pull/1611
- https://github.com/osrf/pounce/pull/22

[Asana Ticket
](https://app.asana.com/0/inbox/1202467295732485/1202257121261053/1202490211162789)

Feature:
- advertises a topic once websocket connection is established
- allows publishing to that topic with the Publish() method

Signed-off-by: Liam Han <liamhan0905@gmail.com>

### CODE REVIEW

File: src/SceneManager.ts
Comment (german-e-mas) [inline src/SceneManager.ts:152]: What's a `Publisher`? I see that this PR point to the `adv_pub` branch. We need a PR of that.

**Review by german-e-mas — COMMENTED** [review]
I'm not sure what the adv_pub branch does. We need a PR from that one.

Other than that, I don't think that having topic, msg data and type here on the scene manager is the best thing to implement this.

File: src/SceneManager.ts
Comment (liamhan0905) [reply src/SceneManager.ts:152]: Sorry for the confusion. So `adv_pub` is the branch that Nate wrote that allows the client to request the server to advertise a public. I saw that `adv_pub` branch is already 65 commits ahead of main. But I think either Nate or you might be the best person to decide whether we should merge `adv_pub` to main.

And here's the definition for the [Publisher](https://github.com/gazebo-web/gzweb/commit/fc86a123e13c60d125dff296a8deac523c7406ef). This is on the `adv_pub` branch.

Comment (german-e-mas) [reply src/SceneManager.ts:152]: Yeah, we should not merge it blindly. I'll take a look, but it will take me more time.

Tagging @coleOsrf for awareness

Comment (liamhan0905) [reply src/SceneManager.ts:152]: Yeah so 
1. We have to cleanup gzweb so that main is somewhat up to date. Cause looking at the branch history, there's been lot of spin offs from `main` but hasn't been merged back. I spun off this PR from Nate's `adv_pub` branch.
2. We could merge this PR to `adv_pub` branch and then merge `adv_pub` to `main` OR first look into merging `adv_pub` into `main` and then merging this PR to `main` as well. Either should be okay.

Comment (german-e-mas) [reply src/SceneManager.ts:152]: We shouldn't aim to `main` but to `gzweb2` - Once `adv_pub` is up to date with `gzweb2` it will be easier to review, but I repeat, we should _not_ merge it.

Comment (liamhan0905) [reply src/SceneManager.ts:152]: Okay that sounds good. Let's first merge `adv_pub` to `gzweb2` and then look at this PR. Do you want me to create a PR for merging `adv_pub` to `gzweb2`? And just to be clear, `gzweb2` is the most stable code that we're using for the hyundai project right?

Comment (german-e-mas) [reply src/SceneManager.ts:152]: Please do not merge `adv_pub` to `gzweb2` - Pull the latest changes of `gzweb2` to `adv_pub` which will make the review process easier.

Comment (german-e-mas) [reply src/SceneManager.ts:152]: The `adv_pub` branch was updated.

Comment (clalancette) [reply src/SceneManager.ts:152]: > The `adv_pub` branch was updated.

Perfect.  If you open a PR with those changes, I can review them and then we can get that into gzweb2.  Then we can rebase this one to point directly to gzweb2.

File: src/SceneManager.ts
Comment (german-e-mas) [inline src/SceneManager.ts:150]: I think `Publisher object to publish to a topic` is a better comment

File: src/SceneManager.ts
Comment (german-e-mas) [inline src/SceneManager.ts:246]: nit: Add a comment explaining what does the method do.

File: src/SceneManager.ts
Comment (german-e-mas) [inline src/SceneManager.ts:247]: Check for the publisher as well.

```suggestion
    if (this.scene && this.publisher) {
```

**Review by german-e-mas — CHANGES_REQUESTED** [review]
Left some comments!

File: src/SceneManager.ts
Comment (german-e-mas) [reply src/SceneManager.ts:152]: The PR is open: https://github.com/gazebo-web/gzweb/pull/29 Once it's merged, this branch will automatically point to `gzweb2`.

**Review by german-e-mas — APPROVED** [review]
LGTM. Again, we can think about multiple publishers once we need them.

**Comment by liamhan0905** [comment]
sounds good. i'll merge this pr then

**Pull request merged**
